datastore/database/get_object_ratings_by_data_id.go

```diff
@@ -0,0 +1,56 @@
+package database
+
+import (
+	"github.com/PretendoNetwork/nex-go/v2"
+	"github.com/PretendoNetwork/nex-go/v2/types"
+	common_globals "github.com/PretendoNetwork/nex-protocols-common-go/v2/globals"
+	datastore_types "github.com/PretendoNetwork/nex-protocols-go/v2/datastore/types"
+)
+
+func GetObjectRatingsByDataID(manager *common_globals.DataStoreManager, dataID types.UInt64) (types.List[datastore_types.DataStoreRatingInfoWithSlot], *nex.Error) {
+	ratings := types.NewList[datastore_types.DataStoreRatingInfoWithSlot]()
+
+	// TODO - If the total value goes above or below int64 min/max, it needs to be rounded to either the mix/max
+	rows, err := manager.Database.Query(`
+		SELECT
+			rs.slot,
+			COALESCE(SUM(r.value), 0) AS total_value,
+			COUNT(r.id) AS rating_count,
+			rs.initial_value
+		FROM
+			datastore.rating_settings rs
```

Comment:
I think this should be like this?

```suggestion
			datastore.rating_settings AS rs
```

Reply:
[`AS` is optional in Postgres](https://www.w3schools.com/postgresql/postgresql_as.php), though I can add it if you would prefer. I can see how `AS` would make it more clear

Reply:
Oh I didn't know about this. Yeah I would add it for clarity